### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by mckaymatt**
The problem with using executor.map is that it was greedy and it would keep calling __next__() on the WorkUnitGenerator even when every thread was busy. 
With this fix we do not call __next__() until a worker thread is finished.

### CODE REVIEW

**Comment by mckaymatt** [comment]
I also removed the manual garbage collection since it didn't seem like it made any difference during my manual testing.

**Comment by mckaymatt** [comment]
@datarobot/predictions  Please review

File: datarobot_batch_scoring/network.py [hunk 1]
```diff
@@ -45,15 +46,23 @@ def _request(self, request):
...
-            del prepared
-            gc.collect()
 
     def perform_requests(self, requests):
-        return self._executor.map(self._request, requests)
+        for r in requests:
+            while True:
+                self.futures = [i for i in self.futures if not i.done()]
+                if len(self.futures) < self.concurrency:
+                    self.futures.append(self._executor.submit(self._request,
```
Comment (pprett) [inline datarobot_batch_scoring/network.py]: my bad - the break below

Comment (mckaymatt) [inline datarobot_batch_scoring/network.py]: We are iterating the WorkUnitGenerator sequentially. The batches generated by the WorkUnitGenerator are then being passed to the executor where they are being executed in threads non-sequentially. The only difference between this approach and the previous approach is that we do not submit additional batches until some of the outstanding threads finish executing. 
https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.submit

**Pull request merged**